pkg/server/commands/connectedobjects/connected_objects.go

```diff
@@ -559,9 +492,32 @@ func (c *ConnectedObjectsQuery) reverseExpandDirect(
 				Relation:         targetObjectRel,
 				User:             sourceUserRef,
 				ContextualTuples: req.contextualTuples,
-			}, resultChan, foundObjectsMap, foundCount)
+			}, resultChan, req.ingress)
 		})
 	}
 
 	return subg.Wait()
 }
+
+func (c *ConnectedObjectsQuery) trySendCandidate(ctx context.Context, ingress *graph.RelationshipIngress, candidateObject string, candidateChan chan<- *ConnectedObjectsResult) {
+	_, span := tracer.Start(ctx, "trySendCandidate", trace.WithAttributes(
+		attribute.String("object", candidateObject),
+		attribute.Bool("sent", false),
+	))
+	defer span.End()
+	if _, ok := c.candidateObjectsMap.LoadOrStore(candidateObject, struct{}{}); !ok {
+		if c.candidatesFound != nil && atomic.AddUint32(c.candidatesFound, 1) > c.maxCandidates {
+			return
+		}
+
+		resultStatus := NoFurtherEvalStatus
+		if ingress != nil && ingress.Condition == graph.RequiresFurtherEvalCondition {
+			resultStatus = RequiresFurtherEvalStatus
+		}
+		candidateChan <- &ConnectedObjectsResult{
+			Object:       candidateObject,
+			ResultStatus: resultStatus,
+		}
+		span.SetAttributes(attribute.Bool("sent", true))
```

Comment:
should we have attribute on resultStatus?

Reply:
Yes, we can